app/src/main/java/org/bepass/oblivion/MainActivity.java

```diff
@@ -47,7 +47,7 @@ private SwitchButton.OnCheckedChangeListener createSwitchCheckedChangeListener()
                 } else {
                     OblivionVpnService.startVpnService(this);
                 }
-            } else if (lastKnownConnectionState == ConnectionState.CONNECTED || lastKnownConnectionState == ConnectionState.CONNECTING) {
+            } else if (lastKnownConnectionState.isConnectedOrConnecting()) {
```

Comment:
This code and the above code would be simpler if there was an `isDisconnected()` method.